lib/registry-sync/ecr-mirror.ts

```diff
@@ -187,14 +185,8 @@ export class EcrRegistrySync extends Construct {
 
   }
 
-  /**
-   * Grant the specified grantees pull privileges to the target ECR repositories.
-   */
-  public grantPull(...grantees: iam.IGrantable[]) {
```

Comment:
Can you explain the reasoning behind removing this? This was rather useful from what I recall.

Reply:
The implementation `grantPull()` grants permissions to all ECR repos, which doesn't follow the 'least privileges' policy.
With the `ecrRepository()` API, it can be easily done by getting the corresponding repo and calling `grantPull` on it.

Reply:
I'm wondering if we should elevate `grantAuthorize()` to the `aws-ecr` module?

Reply:
https://github.com/aws/aws-cdk/pull/11783

Reply:
> aws/aws-cdk#11783

👍 

> The implementation grantPull() grants permissions to all ECR repos, which doesn't follow the 'least privileges' policy.

Ok, I think it's probably ok though to treat the registry as a single grantable unit, especially since its only read, but lets see if this becomes tricky in case the caller doesn't know which repos it needs (granted - not the ideal scenario).